### ISSUE TITLE
[Bug] Qwen3.5-35B-A3B LoRA SFT: loss=0, token_acc=0

### ISSUE THREAD

**Issue by kummar** [modelscope/ms-swift#8142]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

#### Environment
- ms-swift: v4.0.0.dev0 (main branch, pulled 2026-02-28)
- transformers: 5.2.0
- deepspeed: ZeRO-3
- GPU: 4 nodes × 4 L40s (16 GPUs)
- Model: Qwen3.5-35B-A3B

#### Problem

When running LoRA SFT on Qwen3.5-35B-A3B with `swift sft`, loss is always 0.0, token_acc is always 0.0, and grad_norm is stuck at 1.0 (likely only from router_aux_loss).

The same training data works perfectly fine with Qwen3-VL-4B and Qwen3-VL-8B.

Training log (`logging.jsonl`):
```json
{"current_steps": 1, "loss": 0.0, "grad_norm": 1.0, "learning_rate": 0.0, "epoch": 0.0, "token_acc": 0.0}
{"current_steps": 10, "loss": 0.0, "grad_norm": 1.0, "learning_rate": 5.1e-05, "epoch": 0.03, "token_acc": 0.0}
{"current_steps": 20, "loss": 0.0, "grad_norm": 1.0, "learning_rate": 0.0001, "epoch": 0.06, "token_acc": 0.0}
{"current_steps": 30, "loss": 0.0, "grad_norm": 1.0, "learning_rate": 9.9e-05, "epoch": 0.09, "token_acc": 0.0}
```

#### Training Command

swift sft \
  --model /path/to/Qwen3.5-35B-A3B \
  --dataset train_data.jsonl \
  --train_type lora \
  --lora_rank 16 --lora_alpha 32 \
  --deepspeed ds_zero3.json \
  --max_length 12288 \
  --padding_free false --packing false \
  --attn_impl flash_attn
  （loss_scale and add_non_thinking_prefix not specified (using defaults)）

args.json confirms correct model detection:
model_type: qwen3_5_moe
template: qwen3_5
loss_scale: default

#### How to Reproduce

1. Prepare any SFT dataset in standard messages format (with or without images)
2. Run training with default settings:
```bash
swift sft \
  --model Qwen/Qwen3.5-35B-A3B \
  --dataset your_data.jsonl \
  --train_type lora \
  --max_length 12288
3. Observe loss: 0.0, token_acc: 0.0, grad_norm: 1.0 in logging.jsonl

**Comment by Jintao-Huang** [comment]
https://github.com/modelscope/ms-swift/blob/main/examples/models/qwen3_5/transformers.sh

Try this to see if it works properly

**Comment by kummar** [comment]
> https://github.com/modelscope/ms-swift/blob/main/examples/models/qwen3_5/transformers.sh
> 
> Try this to see if it works properly

Thanks for the reply! I compared my config with the official example `examples/models/qwen3_5/transformers.sh` and found several key differences:

**Config differences:**

| Parameter | Official Example | My Config |
|-----------|-----------------|-----------|
| `--add_non_thinking_prefix` | `true` | `false` |
| `--experts_impl` | `grouped_mm` | not set |
| `--attn_impl` | not set (default) | `eager` |
| `--loss_scale` | not set | `all` (for debugging) |

The most likely cause is `--add_non_thinking_prefix false`. I set it to `false` because my use case is **structured information extraction**— the model should directly output structured results without any thinking process at inference time.

My question: what is the correct approach if I don't need thinking during inference?

1. Train with `--add_non_thinking_prefix true`, then disable thinking at inference time via prompt or generation parameters?
2. Or is there another way to disable thinking during training without causing loss=0?
